tests/test_processing.py

```diff
@@ -497,4 +497,4 @@ def test_run_dnatracing(process_scan_config: dict, tmp_path: Path) -> None:
 
     assert isinstance(dnatracing_df, pd.DataFrame)
     assert dnatracing_df.shape[0] == 13
-    assert len(dnatracing_df.columns) == 23
+    assert len(dnatracing_df.columns) == 26
```

Comment:
So previously to this PR, during running of `test_run_dnatracing()`, the lines inside `get_splined_traces()` that have the `.pop()` methods, are called.


<img width="280" alt="image" src="https://github.com/AFM-SPM/TopoStats/assets/86117496/312e53c2-70cc-412b-9aa4-e023aaa4fc8b">


This would cause an AttributeError and the test would fail, except it doesn't since we have a broad `except` around dna tracing in `processing.py`.

<img width="182" alt="image" src="https://github.com/AFM-SPM/TopoStats/assets/86117496/0f5aa084-9ce6-4057-9928-dbed25a5f009">

Dna tracing as a result returns our `create_empty_dataframe()` data frame which doesn't contain the 3 columns for dna tracing data, namely `contour_length`, `circular` and `end_to_end_distance`.

This means that it doesn't crash, and the test runs successfully.
I stupidly didn't check the columns when writing this test, and assumed that since it didn't crash, it worked as intended. (many apologies for this).

So the test currently asserts that the returned data frame has the columns without the dna tracing columns.

I verified this by saving the data frames during the test, here we see that the old one was missing the columns:

Screenshots for ease of seeing the difference:

Old:
<img width="283" alt="image" src="https://github.com/AFM-SPM/TopoStats/assets/86117496/c71c0781-0126-407a-8779-93aae6753881">

New:
<img width="544" alt="image" src="https://github.com/AFM-SPM/TopoStats/assets/86117496/b6b26c6e-ec64-483f-8af2-8d9bee0e5de4">

Copy & Paste of the columns direct from the csv files:


New:
molecule_number,centre_x,centre_y,radius_min,radius_max,radius_mean,radius_median,height_min,height_max,height_median,height_mean,volume,area,area_cartesian_bbox,smallest_bounding_width,smallest_bounding_length,smallest_bounding_area,aspect_ratio,threshold,max_feret,min_feret,image,contour_length,circular,end_to_end_distance,basename


Old:
molecule_number,centre_x,centre_y,radius_min,radius_max,radius_mean,radius_median,height_min,height_max,height_median,height_mean,volume,area,area_cartesian_bbox,smallest_bounding_width,smallest_bounding_length,smallest_bounding_area,aspect_ratio,threshold,max_feret,min_feret,image,basename

Reply:
:+1: Thanks for taking the time to explain all that, I completely agree the broad `try: .... except:` causes more pain that it solves, but there was urgency to get something running that didn't continually fail before refactoring of `dnatracing` was undertaken.

Reply:
Oh I don't deny that using the broad exception should have been used, we needed to get things done 😄